Fix `TestDeletingAppStuckInSync`

```
time="2019-08-02T13:45:26-07:00" level=info msg="kubectl delete ns -l e2e.argoproj.io=true --field-selector status.phase=Active --wait=false" dir= execID=Gqi1w
time="2019-08-02T13:45:26-07:00" level=debug msg="namespace \"argocd-e2e-test-auto-sync-self-heal-enabled-idwtt\" deleted\n" duration=81.567867ms execID=Gqi1w
time="2019-08-02T13:45:26-07:00" level=info msg="Starting configmap/secret informers"
time="2019-08-02T13:45:26-07:00" level=info msg="configmap informer cancelled"
time="2019-08-02T13:45:26-07:00" level=info msg="secrets informer cancelled"
time="2019-08-02T13:45:26-07:00" level=info msg="Configmap/secret informer synced"
time="2019-08-02T13:45:28-07:00" level=info msg="mkdir -p /tmp/argo-e2e" dir= execID=BnY4w
time="2019-08-02T13:45:28-07:00" level=debug duration=3.035298ms execID=BnY4w
time="2019-08-02T13:45:28-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/tls" dir= execID=7gSYy
time="2019-08-02T13:45:28-07:00" level=debug duration=3.029431ms execID=7gSYy
time="2019-08-02T13:45:28-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/ssh" dir= execID=9xlCL
time="2019-08-02T13:45:28-07:00" level=debug duration=2.737901ms execID=9xlCL
time="2019-08-02T13:45:28-07:00" level=info msg="cp -Rf testdata /tmp/argo-e2e/testdata.git" dir= execID=I9qQ5
time="2019-08-02T13:45:28-07:00" level=debug duration=46.211582ms execID=I9qQ5
time="2019-08-02T13:45:28-07:00" level=info msg="chmod 777 ." dir=/tmp/argo-e2e/testdata.git execID=r45DR
time="2019-08-02T13:45:28-07:00" level=debug duration=3.391494ms execID=r45DR
time="2019-08-02T13:45:28-07:00" level=info msg="git init" dir=/tmp/argo-e2e/testdata.git execID=XO9MZ
time="2019-08-02T13:45:28-07:00" level=debug msg="Initialized empty Git repository in /private/tmp/argo-e2e/testdata.git/.git/\n" duration=15.006802ms execID=XO9MZ
time="2019-08-02T13:45:28-07:00" level=info msg="git add ." dir=/tmp/argo-e2e/testdata.git execID=QSejM
time="2019-08-02T13:45:28-07:00" level=debug duration=58.644144ms execID=QSejM
time="2019-08-02T13:45:28-07:00" level=info msg="git commit -q -m initial commit" dir=/tmp/argo-e2e/testdata.git execID=J06Kq
time="2019-08-02T13:45:28-07:00" level=debug duration=42.692262ms execID=J06Kq
time="2019-08-02T13:45:28-07:00" level=info msg="kubectl create ns argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq" dir= execID=0jkxP
time="2019-08-02T13:45:28-07:00" level=debug msg="namespace/argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq created\n" duration=82.140496ms execID=0jkxP
time="2019-08-02T13:45:28-07:00" level=info msg="kubectl label ns argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq e2e.argoproj.io=true" dir= execID=ocUlf
time="2019-08-02T13:45:28-07:00" level=debug msg="namespace/argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq labeled\n" duration=97.858393ms execID=ocUlf
time="2019-08-02T13:45:28-07:00" level=info msg="clean state" duration=1.82301816s id=test-deleting-app-stuck-in-sync-yiqkq name=test-deleting-app-stuck-in-sync
time="2019-08-02T13:45:28-07:00" level=info msg=patching jsonPatch="[{\"op\": \"replace\", \"path\": \"/spec/containers/0/command\", \"value\": [\"sh\", \"-c\", \"until ls /tmp/done; do sleep 0.1; done\"]}]" path=hook/hook.yaml
time="2019-08-02T13:45:28-07:00" level=info msg="converting YAML to JSON"
time="2019-08-02T13:45:28-07:00" level=info msg=JSON bytes="{\"apiVersion\":\"v1\",\"kind\":\"Pod\",\"metadata\":{\"annotations\":{\"argocd.argoproj.io/hook\":\"Sync\"},\"name\":\"hook\"},\"spec\":{\"containers\":[{\"command\":[\"true\"],\"image\":\"alpine:latest\",\"imagePullPolicy\":\"IfNotPresent\",\"name\":\"main\"}],\"restartPolicy\":\"Never\"}}"
time="2019-08-02T13:45:28-07:00" level=info msg="converting JSON back to YAML"
time="2019-08-02T13:45:28-07:00" level=info msg="git diff" dir=/tmp/argo-e2e/testdata.git execID=Khi2W
time="2019-08-02T13:45:28-07:00" level=debug msg="diff --git a/hook/hook.yaml b/hook/hook.yaml\nindex c14b9d2..344c457 100644\n--- a/hook/hook.yaml\n+++ b/hook/hook.yaml\n@@ -6,9 +6,11 @@ metadata:\n   name: hook\n spec:\n   containers:\n-    - command:\n-        - \"true\"\n-      image: \"alpine:latest\"\n-      imagePullPolicy: IfNotPresent\n-      name: main\n-  restartPolicy: Never\n\\ No newline at end of file\n+  - command:\n+    - sh\n+    - -c\n+    - until ls /tmp/done; do sleep 0.1; done\n+    image: alpine:latest\n+    imagePullPolicy: IfNotPresent\n+    name: main\n+  restartPolicy: Never\n" duration=11.953442ms execID=Khi2W
time="2019-08-02T13:45:28-07:00" level=info msg="git commit -am patch" dir=/tmp/argo-e2e/testdata.git execID=8HjT3
time="2019-08-02T13:45:28-07:00" level=debug msg="[master 9f11ad8] patch\n 1 file changed, 8 insertions(+), 6 deletions(-)\n" duration=39.052974ms execID=8HjT3
time="2019-08-02T13:45:28-07:00" level=info msg="../../dist/argocd app create test-deleting-app-stuck-in-sync --repo file:///tmp/argo-e2e/testdata.git --path hook --dest-server https://kubernetes.default.svc --dest-namespace argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq --project default --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=9PG0Y
time="2019-08-02T13:45:28-07:00" level=debug msg="application 'test-deleting-app-stuck-in-sync' created\n" duration=405.900034ms execID=9PG0Y
time="2019-08-02T13:45:28-07:00" level=info msg="polling for expectation" message="found success with message 'application 'test-deleting-app-stuck-in-sync' created'" state=succeeded
time="2019-08-02T13:45:28-07:00" level=info msg="../../dist/argocd app sync test-deleting-app-stuck-in-sync --timeout 5 --async --prune --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=0mP5K
time="2019-08-02T13:45:29-07:00" level=debug duration=105.42696ms execID=0mP5K
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="found success with message ''" state=succeeded
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="operation phase should be Running, is Running" state=succeeded
time="2019-08-02T13:45:29-07:00" level=info msg="../../dist/argocd app delete test-deleting-app-stuck-in-sync --cascade=true --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=XORt7
time="2019-08-02T13:45:29-07:00" level=debug duration=109.166556ms execID=XORt7
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="found success with message ''" state=succeeded
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="operation phase should be Running, is Running" state=succeeded
time="2019-08-02T13:45:29-07:00" level=info msg="../../dist/argocd app terminate-op test-deleting-app-stuck-in-sync --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=uzjlt
time="2019-08-02T13:45:29-07:00" level=debug msg="Application 'test-deleting-app-stuck-in-sync' operation terminating\n" duration=97.409309ms execID=uzjlt
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="found success with message 'Application 'test-deleting-app-stuck-in-sync' operation terminating'" state=succeeded
time="2019-08-02T13:45:29-07:00" level=info msg="kubectl -n argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq exec -i hook touch /tmp/done" dir= execID=BZNiJ
time="2019-08-02T13:45:29-07:00" level=debug duration=109.904204ms execID=BZNiJ
time="2019-08-02T13:45:29-07:00" level=error msg="`kubectl -n argocd-e2e-test-deleting-app-stuck-in-sync-yiqkq exec -i hook touch /tmp/done` failed exit status 1: Error from server (NotFound): pods \"hook\" not found" execID=BZNiJ
time="2019-08-02T13:45:29-07:00" level=info msg="polling for expectation" message="app should not exist" state=pending
time="2019-08-02T13:45:32-07:00" level=info msg="polling for expectation" message="app should not exist" state=pending
time="2019-08-02T13:45:35-07:00" level=info msg="polling for expectation" message="app should not exist" state=pending
time="2019-08-02T13:45:38-07:00" level=info msg="polling for expectation" message="app should not exist" state=pending
time="2019-08-02T13:45:41-07:00" level=info msg="polling for expectation" message="app should not exist" state=pending
--- FAIL: TestDeletingAppStuckInSync (17.76s)
```

--- a/test/e2e/app_deletion_test.go
+++ b/test/e2e/app_deletion_test.go
@@ -3,18 +3,15 @@ package e2e
 import (
 	"testing"
 
+	"github.com/argoproj/argo-cd/errors"
 	. "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
 	. "github.com/argoproj/argo-cd/test/e2e/fixture"
 	. "github.com/argoproj/argo-cd/test/e2e/fixture/app"
-	"github.com/argoproj/argo-cd/test/fixture/test"
 )
 
 // when a app gets stuck in sync, and we try to delete it, it won't delete, instead we must then terminate it
 // and deletion will then just happen
 func TestDeletingAppStuckInSync(t *testing.T) {
-
-	test.Flaky(t)
-
 	Given(t).
 		Async(true).
 		Path("hook").
@@ -25,6 +22,8 @@ func TestDeletingAppStuckInSync(t *testing.T) {
 		Then().
 		// stuck in running state
 		Expect(OperationPhaseIs(OperationRunning)).
+		Expect(SyncStatusIs(SyncStatusCodeSynced)).
+		Expect(ResourceResultNumbering(2)).
 		When().
 		Delete(true).
 		Then().
@@ -34,7 +33,7 @@ func TestDeletingAppStuckInSync(t *testing.T) {
 		TerminateOp().
 		And(func() {
 			// force delete the resource
-			_, _ = Run("", "kubectl", "-n", DeploymentNamespace(), "exec", "-i", "hook", "touch", "/tmp/done")
+			errors.FailOnErr(Run("", "kubectl", "-n", DeploymentNamespace(), "exec", "-i", "hook", "touch", "/tmp/done"))
 		}).
 		Then().
 		// delete is successful

